docs/ops/movement/GatherTree_1.md

```diff
@@ -2,63 +2,60 @@
 
 **Versioned name**: *GatherTree-1*
 
-**Category**: Beam search post-processing
+**Category**: *Data movement*
 
 **Short description**: Generates the complete beams from the ids per each step and the parent beam ids.
 
 **Detailed description**
 
-The GatherTree operation implements the same algorithm as the [GatherTree operation in TensorFlow](https://www.tensorflow.org/addons/api_docs/python/tfa/seq2seq/gather_tree).
+*GatherTree* operation reorders token IDs of a given input tensor `step_id` representing IDs per each step of beam search, based on input tensor `parent_ids` representing the parent beam IDs. For a given beam, past the time step containing the first decoded `end_token` all values are filled in with end_token.
 
-Pseudo code:
+The algorithm in pseudocode is as follows:
 
 ```python
-final_idx[ :, :, :] = end_token
+final_ids[ :, :, :] = end_token
 for batch in range(BATCH_SIZE):
     for beam in range(BEAM_WIDTH):
         max_sequence_in_beam = min(MAX_TIME, max_seq_len[batch])
 
-        parent = parent_idx[max_sequence_in_beam - 1, batch, beam]
+        parent = parent_ids[max_sequence_in_beam - 1, batch, beam]
 
-        final_idx[max_sequence_in_beam - 1, batch, beam] = step_idx[max_sequence_in_beam - 1, batch, beam]
+        final_ids[max_sequence_in_beam - 1, batch, beam] = step_ids[max_sequence_in_beam - 1, batch, beam]
 
         for level in reversed(range(max_sequence_in_beam - 1)):
-            final_idx[level, batch, beam] = step_idx[level, batch, parent]
+            final_ids[level, batch, beam] = step_ids[level, batch, parent]
 
-            parent = parent_idx[level, batch, parent]
+            parent = parent_ids[level, batch, parent]
 
         # For a given beam, past the time step containing the first decoded end_token
         # all values are filled in with end_token.
         finished = False
         for time in range(max_sequence_in_beam):
             if(finished):
-                final_idx[time, batch, beam] = end_token
-            elif(final_idx[time, batch, beam] == end_token):
+                final_ids[time, batch, beam] = end_token
+            elif(final_ids[time, batch, beam] == end_token):
                 finished = True
 ```
 
-Element data types for all input tensors should match each other.
+*GatherTree* operation is equivalent to [GatherTree operation in TensorFlow](https://www.tensorflow.org/addons/api_docs/python/tfa/seq2seq/gather_tree).
```

Comment:
Shouldn't we avoid links to outer sources?

Reply:
My understanding is that we should avoid to **ONLY** refer to 3rd party web-site with explanation. But the algorithm is shortly described and pseudocode provided above. I believe there is no harm on keeping this reference here.